packages/data-context/src/actions/AppActions.ts

```diff
@@ -17,19 +17,24 @@ export interface ApplicationDataApiShape {
 export class AppActions {
   constructor (private ctx: DataContext) {}
 
-  setActiveBrowser (browser: FoundBrowser) {
-    this.ctx.coreData.chosenBrowser = browser
+  async setActiveBrowser (browser: FoundBrowser) {
+    this.ctx.coreData.activeBrowser = browser
+
+    await this.ctx._apis.projectApi.insertProjectPreferencesToCache(this.ctx.lifecycleManager.projectTitle, {
```

Comment:
Reaching so deep in the context object is suspicious.

It could mean that this function is better fitted for the `ProjectActions.ts`